### PR TITLE
Define policy library source name in weave policy agent profile

### PR DESCRIPTION
**Pull request by nohasaayed**

### CODE REVIEW

**Review by alichaddad — APPROVED** [review]

File: charts/weave-policy-agent/values.yaml [hunk 1]
```diff
@@ -42,3 +42,7 @@ policySource:
   # branch:
   # path: ./  # Could be a path to the policies dir or a kustomization.yaml file
   # secretRef: policy-library-auth  # (Optional): Name of the K8s secret with private repo auth credentials
+  # sourceRef: # Could specify a name for an existing GitSource reference instead of creating a new one
+  #   kind: GitRepository
+  #   name: policy-library
+  #   namespace: "{{ .Release.Namespace }}"
```
Comment (alichaddad) [inline charts/weave-policy-agent/values.yaml:48]: This should be a fixed value not a template value

File: charts/weave-policy-agent/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 description: A Weaveworks Helm chart for Kubernetes to configure the policy agent
 name: weave-policy-agent
 appVersion: "1.0.0"
```
Comment (alichaddad) [inline charts/weave-policy-agent/Chart.yaml:4]: Would be better if we upgrade the agent version as well, to keep it up to date

**Review by alichaddad — CHANGES_REQUESTED** [review]

File: charts/weave-policy-agent/values.yaml (hunk 1 above)
Comment (nohasaayed) [reply charts/weave-policy-agent/values.yaml:48]: i am not sure what the fixed value is, didn't we agree to set the namespace the same as the release namespace?

File: charts/weave-policy-agent/Chart.yaml (hunk 2 above)
Comment (nohasaayed) [reply charts/weave-policy-agent/Chart.yaml:4]: the agent release blocked on token permission, would upgrade the version here once the agent new version is released.
Not going to merge this PR until the agent version is upgraded

File: charts/weave-policy-agent/values.yaml (hunk 1 above)
Comment (alichaddad) [reply charts/weave-policy-agent/values.yaml:48]: You can't use template values in the `values` file as far as I know. We can use a value like `flux-system`

Comment (nohasaayed) [reply charts/weave-policy-agent/values.yaml:48]: no you could use it as long las you're using `tpl` function in the template

Comment (ahsayde) [reply charts/weave-policy-agent/values.yaml:48]: i prefer to make it a placeholder, for example :
```yaml
  #   name: < source name >
  #   namespace:  < source namespace >
```

Comment (nohasaayed) [reply charts/weave-policy-agent/values.yaml:48]: @alichaddad @ahsayde but as i remember we decided to set the default values here

Comment (alichaddad) [reply charts/weave-policy-agent/values.yaml:48]: We already have a default value for the name and kind and it makes sense to follow that in the namespace, using a template value makes it a bit confusing

Comment (nohasaayed) [reply charts/weave-policy-agent/values.yaml:48]: okay so will use `flux-system` as the default. @ahsayde any concerns?

Comment (ahsayde) [reply charts/weave-policy-agent/values.yaml:48]: @nohasaayed no it's ok for me

**Review by ahsayde — APPROVED** [review]

**Review by waleedhammam — APPROVED** [review]

**Review by alichaddad — APPROVED** [review]

**Pull request merged**